src/services/textChanges.ts

```diff
@@ -291,11 +297,8 @@ namespace ts.textChanges {
             return this;
         }
 
-        // TODO (https://github.com/Microsoft/TypeScript/issues/21246): default should probably be useNonAdjustedPositions
-        public replaceNode(sourceFile: SourceFile, oldNode: Node, newNode: Node, options: ChangeNodeOptions = {}) {
-            const pos = getAdjustedStartPosition(sourceFile, oldNode, options, Position.Start);
-            const end = getAdjustedEndPosition(sourceFile, oldNode, options);
-            return this.replaceRange(sourceFile, { pos, end }, newNode, options);
+        public replaceNode(sourceFile: SourceFile, oldNode: Node, newNode: Node, options: InsertNodeOptions = {}) {

```

Comment:
I was only expecting the default to change but this looks like it makes the alternative (i.e. adjusting start and end positions) impossible.  Was that intentional?

Reply:
Also it seems strange to update this signature without making corresponding changes to (e.g.) `replaceNodeRange`.

Reply:
I don't think we're likely to have any reason to replace the a node's comments while replacing it -- if we do end up needing that, we can add it back.

Reply:
Updated `replaceNodeRange`

Reply:
I guess I think of replace(Node | Range | NodeRange)(WithNodes)? as a family of related functions that should have parallel signatures.  Notice that the WithNodes variants already default to using non-adjusted positions, which should make things easier.

Reply:
Added the option back where it makes sense, but not for `replaceRange` which should only accept `InsertNodeOptions` -- it doesn't adjust pos and end since it accepts a range.

Reply:
Created #22813 which will be easier to review independently, then can merge into this.